### ISSUE TITLE
[concept] AI Copilot

### ISSUE THREAD

**Issue by melvincarvalho** [ondras/my-mind#174]
Currently the idea of copilot in tools are gaining popularity

I was wondering what mind mapping folks think about this

Below is a video form [https://www.superusapp.com/](superus) which demonstrates the concept

https://www.superusapp.com/chatgpt2d/assets/img/chatgpt2d-muted.mp4

Another is a [todo list that does itself for you](https://www.doanythingmachine.com/)

Personally when I run my-mind locally I set up a process called, aimon.sh, which monitors changes in a map that I make when I hit save (saved to [nosDAV](https://nosdav.com) and it will

- run a command
- give myself a payment reward
- use AI to expand questions
- use [NosDAV pubsub](https://github.com/nosdav/nosdav-pubsub#components) to replace firebase with an open system for realtime, collaborative, payments enabled maps

I dont yet have todolists that do themselves, but I think that would be kind of cool, and head towards AGI for mind maps.

I am wondering if there is an appetite for any of these features.

**Comment by arkin0x** [comment]
![invader-zim-mighty-need](https://user-images.githubusercontent.com/99223753/234628481-9ae411cc-bc0f-4550-afb3-016177fea195.gif)

**Comment by rquast** [comment]
I am considering modifying my-mind to muck around with anythingllm which offers a pretty easy to use API for this purpose.

**Comment by rquast** [comment]
I reckon I could probably make some kind of plugin that hooks into subscribers in pubsub.ts (e.g. "item-change" events) to listen for changes to nodes. I'd probably keep it in a separate project and not pollute the lightweight code here. Thinking I might expose expose subscribers on a global scope rather than a module scope so I can access it from another script without needing to create some kind of plugin system.

**Comment by melvincarvalho** [comment]
There already is such code in there for firebase updates.  However firebase is proprietary.

I prefer a pub/sub solution over websockets called nostr:

https://nostr.how/en/what-is-nostr

Which does pub/sub over websockets.  More background on nostr [here](https://www.youtube.com/watch?v=aA-jiiepOrE).  And it might be possible to apply for an open source grant to integrate my mind and nostr [here](https://opensats.org/funds/nostr)

**Comment by rquast** [comment]
Yeah, I can see this in the firebase one:

pubsub.subscribe("firebase-list", this);
pubsub.subscribe("firebase-change", this);

...

case "firebase-change":
if (data) {
	pubsub.unsubscribe("item-change", this);
	app.currentMap.mergeWith(data);
	pubsub.subscribe("item-change", this);
} else { /* FIXME */
	console.log("remote data disappeared");
}

That's a good place to start. Thanks.

**Comment by rquast** [comment]
Yeah I like the idea of your nostr pub/sub backend. Probably go with something like that to keep it easy and use the firebase code for a bit of help.

**Comment by melvincarvalho** [comment]
@rquast awesome!  I would be happy to run a nostr relay if you want to test.  But there are a few 100 [here](https://nostr.watch/relays/find)

A nostr relay also be run with a single line of code if you have docker:

```bash
# docker pull scsibug/nostr-rs-relay
docker run -p 4445:8080 scsibug/nostr-rs-relay
```
